### ISSUE TITLE
Writing XLSX format and column/alignment problem

### ISSUE THREAD

**Issue by finalwebsites** [box/spout#122]
Hi,

Great script, thanks for sharing your code!
I have a problem while writing XLSX files. Some files doesn't look okay while opening them in open office. The column width is fixed and you can't optimize the width (f.e. double click on the right border of the column head in Open office). The content is aligned to the right and not left. So the data looks okay, but you can't use the file very well. It's very random only some columns behave like that.
If you open the example you will see what I mean.

https://www.dropbox.com/s/1ok7kq3swmboqz4/test_wrong.xlsx?dl=0

Is this a know issue?

PS. Strange is that I can't upload any XLSX file to this form (error: Something went really wrong, and we can’t process that file.), maybe not totally valid?

regards
Olaf

**Comment by adrilo** [comment]
Hey @finalwebsites,

There seems to be something wrong with the "wrap text" feature when reading an XLSX file in Open Office. Although if you unset and re-set "wrap text" on the column, it fixes the weird display.
Have you tried exporting an ODS file instead to see if that works better?

In Excel, it works as expected (although the default width for some columns is way too small... I need to fix that!).

**Comment by finalwebsites** [comment]
Hi @adrilo,
Thanks for the tips.
You're right a re-set on the wrapping option is good workarround and the files are okay in MS Excel and Google Spreadsheets. No I never tried the .ods file.

**Comment by adrilo** [comment]
I'd like to keep this open, as this is an actual issue.

**Comment by adrilo** [comment]
I investigated deeper and the only way to have this properly working is by setting the exact height of the row. Which requires intensive calculations: for each cell, based on the chosen font and the width of the column, we need to send the text content to a graphic library to get what the final height of the row needs to be to have the whole text fit.

To keep good performance, I'd like to avoid doing this if possible. Anyway, I created a new issue to track this enhancement, if I (or someone else) ever wants to tackle it!
